### PR TITLE
add list API key Operations

### PR DESCRIPTION
**Pull request by illi-j**
This PR add the command list Operations for an API key.

Usage:
```
./exo lab iam apikey operations -h
This command lists all suported Operations for an API key.

	Supported output template annotations: .Compute, .DNS, .IAM, .SOS

Usage:
  exo lab iam apikey operations [filter ...] [flags]

Flags:
  -h, --help   help for operations

Global Flags:
  -C, --config string            Specify an alternate config file [env EXOSCALE_CONFIG]
  -O, --output-format string     Output format (table|json|text)
      --output-template string   Template to use if output format is "text"
  -Q, --quiet                    Quiet mode (disable non-essential command output
  -A, --use-account string       Account to use in config file [env EXOSCALE_ACCOUNT]
```

List operations with filter:
```
./exo lab iam apikey operations "iam" "dns"
┼─────────┼───────────────────────────┼
│ Compute │ n/a                       │
│ DNS     │ dns/*                     │
│         │ dns/creatednsdomain       │
│         │ dns/creatednsdomainrecord │
│         │ dns/deletednsdomain       │
│         │ dns/deletednsdomainrecord │
│         │ dns/getdnsdomainrecord    │
│         │ dns/getdnsdomainzonefile  │
│         │ dns/listdnsdomainrecords  │
│         │ dns/listdnsdomains        │
│         │ dns/resetdnsdomaintoken   │
│         │ dns/updatednsdomainrecord │
│ IAM     │ iam/*                     │
│         │ iam/createapikey          │
│         │ iam/getapikey             │
│         │ iam/listapikeyoperations  │
│         │ iam/listapikeys           │
│         │ iam/revokeapikey          │
│ SOS     │ n/a                       │
┼─────────┼───────────────────────────┼
```

List all operations:
```
./exo lab iam apikey operations 
┼─────────┼────────────────────────────────────────┼
│ Compute │ compute/*                              │
│         │ compute/activateIp6                    │
│         │ compute/addIpToNic                     │
│         │ compute/addNicToVirtualMachine         │
│         │ compute/associateIpAddress             │
│         │ compute/attachIso                      │
│         │ compute/attachVolume                   │
│         │ compute/authorizeSecurityGroupEgress   │
│         │ compute/authorizeSecurityGroupIngress  │
│         │ compute/changeServiceForVirtualMachine │
│         │ compute/createAffinityGroup            │
│         │ compute/createInstanceGroup            │
│         │ compute/createInstancePool             │
│         │ compute/createNetwork                  │
│         │ compute/createSSHKeyPair               │
│         │ compute/createSecurityGroup            │
│         │ compute/createSnapshot                 │
│         │ compute/createTags                     │
│         │ compute/deleteAffinityGroup            │
│         │ compute/deleteInstanceGroup            │
│         │ compute/deleteNetwork                  │
│         │ compute/deleteSSHKeyPair               │
│         │ compute/deleteSecurityGroup            │
│         │ compute/deleteSnapshot                 │
│         │ compute/deleteTags                     │
│         │ compute/deleteTemplate                 │
│         │ compute/deleteVolume                   │
│         │ compute/deployVirtualMachine           │
│         │ compute/destroyInstancePool            │
│         │ compute/destroyVirtualMachine          │
│         │ compute/detachIso                      │
│         │ compute/detachVolume                   │
│         │ compute/disassociateIpAddress          │
│         │ compute/expungeVirtualMachine          │
│         │ compute/extractVolume                  │
│         │ compute/getCloudIdentifier             │
│         │ compute/getInstancePool                │
│         │ compute/getVMPassword                  │
│         │ compute/getVirtualMachineUserData      │
│         │ compute/listAccounts                   │
│         │ compute/listAffinityGroupTypes         │
│         │ compute/listAffinityGroups             │
│         │ compute/listApis                       │
│         │ compute/listAsyncJobs                  │
│         │ compute/listCapabilities               │
│         │ compute/listDiskOfferings              │
│         │ compute/listEventTypes                 │
│         │ compute/listEvents                     │
│         │ compute/listInstanceGroups             │
│         │ compute/listInstancePools              │
│         │ compute/listIsoPermissions             │
│         │ compute/listIsos                       │
│         │ compute/listNetworkOfferings           │
│         │ compute/listNetworks                   │
│         │ compute/listNics                       │
│         │ compute/listOsCategories               │
│         │ compute/listOsTypes                    │
│         │ compute/listPciDevices                 │
│         │ compute/listPciVendors                 │
│         │ compute/listPublicIpAddresses          │
│         │ compute/listRegions                    │
│         │ compute/listResourceDetails            │
│         │ compute/listResourceLimits             │
│         │ compute/listSSHKeyPairs                │
│         │ compute/listSecurityGroups             │
│         │ compute/listServiceOfferings           │
│         │ compute/listSnapshotPolicies           │
│         │ compute/listSnapshots                  │
│         │ compute/listTags                       │
│         │ compute/listTemplatePermissions        │
│         │ compute/listTemplates                  │
│         │ compute/listTemplatesForUsage          │
│         │ compute/listVirtualMachines            │
│         │ compute/listVolumes                    │
│         │ compute/listZones                      │
│         │ compute/queryAsyncJobResult            │
│         │ compute/rebootVirtualMachine           │
│         │ compute/registerCustomTemplate         │
│         │ compute/registerPrivateTemplate        │
│         │ compute/registerSSHKeyPair             │
│         │ compute/registerUserKeys               │
│         │ compute/removeIpFromNic                │
│         │ compute/removeNicFromVirtualMachine    │
│         │ compute/resetPasswordForVirtualMachine │
│         │ compute/resetSSHKeyForVirtualMachine   │
│         │ compute/resizeVolume                   │
│         │ compute/restoreVirtualMachine          │
│         │ compute/revertSnapshot                 │
│         │ compute/revokeSecurityGroupEgress      │
│         │ compute/revokeSecurityGroupIngress     │
│         │ compute/scaleInstancePool              │
│         │ compute/scaleVirtualMachine            │
│         │ compute/startVirtualMachine            │
│         │ compute/stopVirtualMachine             │
│         │ compute/updateInstanceGroup            │
│         │ compute/updateInstancePool             │
│         │ compute/updateIpAddress                │
│         │ compute/updateNetwork                  │
│         │ compute/updateTemplate                 │
│         │ compute/updateVirtualMachine           │
│         │ compute/updateVmNicIp                  │
│ DNS     │ dns/*                                  │
│         │ dns/createDnsDomain                    │
│         │ dns/createDnsDomainRecord              │
│         │ dns/deleteDnsDomain                    │
│         │ dns/deleteDnsDomainRecord              │
│         │ dns/getDnsDomainRecord                 │
│         │ dns/getDnsDomainZoneFile               │
│         │ dns/listDnsDomainRecords               │
│         │ dns/listDnsDomains                     │
│         │ dns/resetDnsDomainToken                │
│         │ dns/updateDnsDomainRecord              │
│ IAM     │ iam/*                                  │
│         │ iam/createApiKey                       │
│         │ iam/getApiKey                          │
│         │ iam/listApiKeyOperations               │
│         │ iam/listApiKeys                        │
│         │ iam/revokeApiKey                       │
│ SOS     │ sos/*                                  │
┼─────────┼────────────────────────────────────────┼
```

### CODE REVIEW

File: cmd/iam_apikey_operations.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	
+	Supported output template annotations: %s`,
+		strings.Join(outputterTemplateAnnotations(&apiKeyOperationsItemOutput{}), ", ")),
+	Aliases: gShowAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return output(listApiKeyOperations(args))
+	},
+}
+
+func listApiKeyOperations(filters []string) (outputter, error) {
```
Comment (pierre-emmanuelJ) [inline cmd/iam_apikey_operations.go]: Should be `listAPIKeyOperations`

**Comment by pierre-emmanuelJ** [comment]
What about an other way of implementation like:

```Golang
func listAPIKeyOperations(filters []string) (outputter, error) {
	resp, err := cs.RequestWithContext(gContext, &egoscale.ListAPIKeyOperations{})
	if err != nil {
		return nil, err
	}

	opes := resp.(*egoscale.ListAPIKeyOperationsResponse)
	out := apiKeyOperationsItemOutput{}

	for _, o := range opes.Operations {
		operation := strings.ToLower(o)

		result := operation
		for _, f := range filters {
			result = ""
			filter := strings.ToLower(f)
			if strings.Contains(operation, filter) {
				result = operation
				break
			}
		}

		switch true {
		case strings.HasPrefix(result, "compute"):
			out.Compute = append(out.Compute, result)
		case strings.HasPrefix(result, "dns"):
			out.DNS = append(out.DNS, result)
		case strings.HasPrefix(result, "iam"):
			out.IAM = append(out.IAM, result)
		case strings.HasPrefix(result, "sos"):
			out.SOS = append(out.SOS, result)
		}
	}

	return &out, nil
}

```

**Comment by pierre-emmanuelJ** [comment]
IMO do you think a `list` subcommand is better?

File: cmd/iam_apikey_operations.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+func (o *apiKeyOperationsItemOutput) toJSON()  { outputJSON(o) }
+func (o *apiKeyOperationsItemOutput) toText()  { outputText(o) }
+func (o *apiKeyOperationsItemOutput) toTable() { outputTable(o) }
+
+// apiKeyShowCmd represents the API key showing command
+var apiKeyOperationsCmd = &cobra.Command{
+	Use:   "operations [Filter ...]",
+	Short: "List Operations",
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]:
```suggestion
	Short: "List supported API key operations",
```

File: cmd/iam_apikey_operations.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	DNS     []string `json:"dns,omitempty"`
+	IAM     []string `json:"iam,omitempty"`
+	SOS     []string `json:"sos,omitempty"`
+}
+
+func (o *apiKeyOperationsItemOutput) toJSON()  { outputJSON(o) }
+func (o *apiKeyOperationsItemOutput) toText()  { outputText(o) }
+func (o *apiKeyOperationsItemOutput) toTable() { outputTable(o) }
+
+// apiKeyShowCmd represents the API key showing command
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]: Comment doesn't match function.

File: cmd/iam_apikey_operations.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+var apiKeyOperationsCmd = &cobra.Command{
+	Use:   "operations [Filter ...]",
+	Short: "List Operations",
+	Long: fmt.Sprintf(`This command lists all Operations for an API key.
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]:
```suggestion
	Long: fmt.Sprintf(`This command lists all supported operations for an API key.
```

File: cmd/iam_apikey_operations.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	SOS     []string `json:"sos,omitempty"`
+}
+
+func (o *apiKeyOperationsItemOutput) toJSON()  { outputJSON(o) }
+func (o *apiKeyOperationsItemOutput) toText()  { outputText(o) }
+func (o *apiKeyOperationsItemOutput) toTable() { outputTable(o) }
+
+// apiKeyShowCmd represents the API key showing command
+var apiKeyOperationsCmd = &cobra.Command{
+	Use:   "operations [Filter ...]",
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]:
```suggestion
	Use:   "operations [filter ...]",
```

Please document the `filter` argument in the command long help field.

**Comment by illi-j** [comment]
> What do you think about an other way of implementation 😉?
> 
> More like:
> 
> ```go
> func listAPIKeyOperations(filters []string) (outputter, error) {
> 	resp, err := cs.RequestWithContext(gContext, &egoscale.ListAPIKeyOperations{})
> 	if err != nil {
> 		return nil, err
> 	}
> 
> 	opes := resp.(*egoscale.ListAPIKeyOperationsResponse)
> 	out := apiKeyOperationsItemOutput{}
> 
> 	for _, o := range opes.Operations {
> 		operation := strings.ToLower(o)
> 
> 		result := operation
> 		for _, f := range filters {
> 			result = ""
> 			filter := strings.ToLower(f)
> 			if strings.Contains(operation, filter) {
> 				result = operation
> 				break
> 			}
> 		}
> 
> 		switch true {
> 		case strings.HasPrefix(result, "compute"):
> 			out.Compute = append(out.Compute, result)
> 		case strings.HasPrefix(result, "dns"):
> 			out.DNS = append(out.DNS, result)
> 		case strings.HasPrefix(result, "iam"):
> 			out.IAM = append(out.IAM, result)
> 		case strings.HasPrefix(result, "sos"):
> 			out.SOS = append(out.SOS, result)
> 		}
> 	}
> 
> 	return &out, nil
> }
> ```

Nice ! thx 👍

File: cmd/iam_apikey_operations.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+func (o *apiKeyOperationsItemOutput) toJSON()  { outputJSON(o) }
+func (o *apiKeyOperationsItemOutput) toText()  { outputText(o) }
+func (o *apiKeyOperationsItemOutput) toTable() { outputTable(o) }
+
+// apiKeyOperationsCmd represents the supported operations listing command for an API key
+var apiKeyOperationsCmd = &cobra.Command{
+	Use:   "operations [filter ...]",
+	Short: "List supported API key operations",
+	Long: fmt.Sprintf(`This command lists all suported Operations for an API key.
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]:
```suggestion
	Long: fmt.Sprintf(`This command lists all supported operations for an API key.
```

File: cmd/iam_apikey_operations.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	SOS     []string `json:"sos,omitempty"`
+}
+
+func (o *apiKeyOperationsItemOutput) toJSON()  { outputJSON(o) }
+func (o *apiKeyOperationsItemOutput) toText()  { outputText(o) }
+func (o *apiKeyOperationsItemOutput) toTable() { outputTable(o) }
+
+// apiKeyOperationsCmd represents the supported operations listing command for an API key
+var apiKeyOperationsCmd = &cobra.Command{
+	Use:   "operations [filter ...]",
```
Comment (falzm) [inline cmd/iam_apikey_operations.go]: The `filter` argument is still missing documentation.

**Pull request merged**